### PR TITLE
[ETFE-2479]: added DocumentsAddToList page/controller/routes and added indexes Documents section pages 

### PR DESCRIPTION
**Pull request by LadonJackson**

### CODE REVIEW

File: app/controllers/sections/documents/DocumentsAddToListController.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          .set(DocumentsAddToListPage, DocumentsAddToList.Yes)
+
+        userAnswersService.set(request.userAnswers.remove(DocumentsAddToListPage))
+          .map { _ =>
+            Redirect(navigator.nextPage(
+              page = DocumentsAddToListPage,
+              mode = NormalMode,
+              userAnswers = answersWithDocumentAddToList
+            ))
+          }
```
Comment (MJCallahanPage) [inline app/controllers/sections/documents/DocumentsAddToListController.scala:88]: I'm a little confused by this, why does it set this to `Yes` but then remove it when it stores it?

File: app/navigation/DocumentsNavigator.scala [hunk 2]
```diff
@@ -86,13 +63,41 @@ class DocumentsNavigator @Inject() extends BaseNavigator {
       reviewRouteMap(page)(userAnswers)
   }
 
-  private def referenceAvailableRouting(mode: Mode = NormalMode): UserAnswers => Call = (userAnswers: UserAnswers) =>
-    userAnswers.get(ReferenceAvailablePage) match {
+  private def documentsCertificatesRouting(mode: Mode = NormalMode): UserAnswers => Call = (userAnswers: UserAnswers) =>
+    userAnswers.get(DocumentsCertificatesPage) match {
+      case Some(false) => routes.DocumentsCheckAnswersController.onPageLoad(userAnswers.ern, userAnswers.draftId)
+      case _ => userAnswers.get(DocumentsCount) match {
+        case Some(0) | None => routes.DocumentTypeController.onPageLoad(userAnswers.ern, userAnswers.draftId, 0, NormalMode)
```
Comment (MJCallahanPage) [inline app/navigation/DocumentsNavigator.scala:70]: Wouldn't this be `1` ? To add the first one?

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
Couple of queries

File: app/navigation/DocumentsNavigator.scala (hunk 2 above)
Comment (LadonJackson) [reply app/navigation/DocumentsNavigator.scala:70]: If I'm understanding this correctly, 0 is the first one right?

File: app/controllers/sections/documents/DocumentsAddToListController.scala (hunk 1 above)
Comment (MJCallahanPage) [reply app/controllers/sections/documents/DocumentsAddToListController.scala:88]: Makes sense 👍

File: app/navigation/DocumentsNavigator.scala (hunk 2 above)
Comment (MJCallahanPage) [reply app/navigation/DocumentsNavigator.scala:70]: Ah yeah, missed that this gets implicitly converted into a Index model. 👍

**Review by MJCallahanPage — APPROVED** [review]
LGTM 🟢 🚀

**Pull request merged**